src/core/simulation_results.jl

```diff
@@ -1,517 +1,420 @@
-variable = [:P__ThermalStandard, :P__RenewableDispatch]
-results = load_simulation_results(stage,step, variable, SimulationResultsReference)
-```
-# Accepted Key Words
-"""
+    Returns the values for the requested variable names. Accepts a vector of names for the
+    return of the values. If the time stamps and names are loaded using the [load_simulation_results!](@ref)
+    function it will read from memory.
 
-function load_simulation_results(
-    path::String,
-    stage_name::String,
-    step::Array,
-    variable::Array;
-    kwargs...,
+    # Accepted Key Words
+    - `initial_time::Dates.DateTime` : initial of the requested results
+    - `count::Int`: Number of results
+"""
+function get_variables_values(
+    res::SimulationResults,
+    names::Vector{Symbol};
+    initial_time::Union{Nothing, Dates.DateTime} = nothing,
+    count::Union{Int, Nothing} = nothing,
 )
-    sim_results = deserialize_sim_output(path)
-    load_simulation_results(sim_results, stage_name, step, variable; kwargs...)
+    timestamps = _process_timestamps(res, initial_time, count)
+    values = _get_variables_values(res, names, timestamps)
+    return values
 end
 
-function load_simulation_results(
-    sim_output::SimulationResultsReference,
-    stage_name::String,
-    step::Array,
-    variable::Array;
-    kwargs...,
-)
-    results_folder = mkpath(joinpath(sim_output.results_folder, stage_name))
-    stage = "stage-$stage_name"
-    references = sim_output.ref
-    base_power = sim_output.base_powers[stage_name]
-    variables = Dict{Symbol, DataFrames.DataFrame}()
-    duals = Dict{Symbol, Any}()
-    params = Dict{Symbol, DataFrames.DataFrame}()
-    time_stamp = DataFrames.DataFrame(Range = Dates.DateTime[])
-    time_length = sim_output.chronologies[stage]
-    dual = _find_duals(collect(keys(references[stage])))
-    param = _find_params(collect(keys(references[stage])))
-    variable = setdiff(variable, vcat(param, dual))
-    for l in 1:length(variable)
-        date_df = references[stage][variable[l]]
-        step_df = DataFrames.DataFrame(
-            Date = Dates.DateTime[],
-            Step = String[],
-            File_Path = String[],
-        )
-        for n in 1:length(step)
-            step_df = vcat(step_df, date_df[date_df.Step .== step[n], :])
-        end
-        variables[(variable[l])] = DataFrames.DataFrame()
-        for (ix, time) in enumerate(step_df.Date)
-            file_path = step_df[ix, :File_Path]
-            var = read_arrow_file("$file_path")
-            variables[(variable[l])] = vcat(variables[(variable[l])], var[1:time_length, :])
-            if l == 1
-                time_stamp = vcat(time_stamp, _read_time(file_path, time_length))
-                check_file_integrity(dirname(file_path))
-            end
-        end
+function _get_duals_values(res::SimulationResults, names::Vector{Symbol}, timestamps)
+    isempty(names) && return res.dual_values
```

Comment:
Having the user pass an empty vector of names seems slightly awkward. Should there be another function that doesn't accept `names`?

Reply:
this function is internal, so I can use the same code for load_simulation_results! and get_variable_results. The request could be empty.

Reply:
This function is internal but the user could call `get_duals_values(res, []Symbol)` and get cached results. You didn't document that, but it would work. You could formalize that interface by having another method that doesn't have the names parameter.